Assets/BidMachine/Api/UserPermissions.cs

```diff
@@ -3,37 +3,33 @@
 
 namespace BidMachineAds.Unity.Api
 {
+    [System.Obsolete("This class is deprecated and will be removed in future versions")]
     public sealed class UserPermissions : IUserPermissions
     {
-        private readonly IUserPermissions client;
+        public UserPermissions() { }
 
-        public UserPermissions()
-        {
-            this.client = BidMachineClientFactory.GetUserPermissions();
-        }
-
-        public UserPermissions(IUserPermissions client)
-        {
-            this.client = client;
-        }
+        public UserPermissions(IUserPermissions client) { }
 
         /// <summary>
         /// Check CoarseLocation and FineLocation permission.
         /// See <see cref="BidMachine.CheckAndroidPermissions"/> for resulting triggered event.
         /// <param name="permission">Permission.CoarseLocation or Permission.CoarseLocation</param>
         /// </summary>
+        [System.Obsolete("Check method is deprecated and will be removed in future versions")]
```

Comment:
If I am not mistaken: if class is marked as obsolete - all methods, properties, and fields of that class will automatically be considered obsolete as well

Reply:
✅